### PR TITLE
CI: adding cython coverage

### LINKED ISSUE #43652: CI: codecov for pyx files
#### Is your feature request related to a problem?

cython supports [line coverage](https://cython.readthedocs.io/en/latest/src/tutorial/profiling_tutorial.html#enabling-coverage-analysis) but pyx files are currently not included in the codecov reports. Having code coverage for cython code might help to remove unused cython code.

#### Describe the solution you'd like

pyx files are included in the codecov report.

### PR DESCRIPTION
**Pull request by debnathshoham**
- [x] closes #43652

Trying to include cython codes in codecov reports.
Working with the CI for the first time.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[run]
```
Comment (jbrockmendel) [inline .coveragerc:1]: can this go in the coverage section of setup.cfg?

File: ci/setup_env.sh [hunk 2]
```diff
@@ -103,7 +103,7 @@ conda list pandas
 # Make sure any error below is reported as such
 
 echo "[Build extensions]"
-python setup.py build_ext -q -j2
+python setup.py build_ext -q -j2 --with-cython-coverage
```
Comment (jbrockmendel) [inline ci/setup_env.sh:106]: im pretty sure we dont want this on every build, since it slows things down a lot.

Comment (debnathshoham) [reply ci/setup_env.sh:106]: makes sense.. would you know why the build is failing though?

Comment (jbrockmendel) [reply ci/setup_env.sh:106]: pandas/_libs/algos.c: In function ‘__pyx_fuse_0__pyx_f_6pandas_5_libs_5algos_rank_sorted_1d.isra.0.constprop’:
pandas/_libs/algos.c:1023:34: error: ‘__pyx_gilstate_save’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
 1023 | #define __Pyx_PyGILState_Release PyGILState_Release
      |                                  ^~~~~~~~~~~~~~~~~~
pandas/_libs/algos.c:73649:20: note: ‘__pyx_gilstate_save’ was declared here
73649 |   PyGILState_STATE __pyx_gilstate_save;

File: .coveragerc (hunk 1 above)
Comment (debnathshoham) [reply .coveragerc:1]: tried that in another repo. it has it be in `.coveragerc`

File: .github/workflows/python-cython.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+name: Python build with Cython
```
Comment (jreback) [inline .github/workflows/python-cython.yml:1]: umm we need another build? can you do it in an existing one

**Review by jreback — CHANGES_REQUESTED** [review]

File: .coveragerc (hunk 1 above)
Comment (jbrockmendel) [reply .coveragerc:1]: OK, if this file is re-introduced, then probably should move the coverage stuff currently in setup.cfg (which currently includes this Cython.Coverage line)

Comment (debnathshoham) [reply .coveragerc:1]: you're right. my bad!

**Comment by alimcmaster1** [comment]
@debnathshoham - do you still want to work on this? See comments above - H2H if needed

**Comment by jreback** [comment]
closing as stale

**Pull request closed without merging**